src/structure/model.jl

```diff
@@ -663,463 +956,658 @@ Base.eltype(::Type{Residue}) = AbstractAtom
+        'H' in atomname(at) &&
+        !ismatch(r"[a-zA-Z]", atomname(at)[1:findfirst(atomname(at), 'H')-1]))
+end
+
+
+# Residues are ordered in a chain with all hetero residues at the end
+# For obtaining sequence we will re-order them numerically
+function AminoAcidSequence(ch::Chain, residue_selectors::Function...)
+    return AminoAcidSequence(
+        sort(collectresidues(ch, residue_selectors...), by=resnumber))
+end
+
+function AminoAcidSequence{T <: AbstractResidue}(res::Vector{T})
+    seq = AminoAcid[]
+    for i in 1:length(res)
+        if resname(res[i]) in keys(threeletter_to_aa)
+            push!(seq, threeletter_to_aa[resname(res[i])])
+        else
+            push!(seq, AA_X)
+        end
+        if i+1 <= length(res) && resnumber(res[i+1]) - resnumber(res[i]) > 1
+            append!(seq, [AA_Gap for _ in 1:(resnumber(res[i+1]) - resnumber(res[i]) - 1)])
+        end
+    end
+    return AminoAcidSequence(seq)
+end
 
 
 # Descriptive showing of elements
 
 function Base.show(io::IO, struc::ProteinStructure)
     if countmodels(struc) > 0
-        model = defaultmodel(struc)
+        mod = defaultmodel(struc)
         println(io, "Name                        -  ", structurename(struc))
         println(io, "Number of models            -  ", countmodels(struc))
-        println(io, "Chain(s)                    -  ", join(chainids(model)))
-        println(io, "Number of residues          -  ", countresidues(model, stdresselector))
-        println(io, "Number of point mutations   -  ", countresidues(model, stdresselector, disorderselector))
-        println(io, "Number of other molecules   -  ", countresidues(model, hetresselector) - countresidues(model, hetresselector, waterselector))
-        println(io, "Number of water molecules   -  ", countresidues(model, hetresselector, waterselector))
-        println(io, "Number of atoms             -  ", countatoms(model, stdatomselector))
-        println(io, "Number of hydrogens         -  ", countatoms(model, stdatomselector, hydrogenselector))
-        println(io, "Number of disordered atoms  -  ", countatoms(model, stdatomselector, disorderselector))
+        println(io, "Chain(s)                    -  ", join(chainids(mod)))
+        println(io, "Number of residues          -  ", countresidues(mod, stdresselector))
+        println(io, "Number of point mutations   -  ", countresidues(mod, stdresselector, disorderselector))
+        println(io, "Number of other molecules   -  ", countresidues(mod, hetresselector) - countresidues(mod, hetresselector, waterselector))
+        println(io, "Number of water molecules   -  ", countresidues(mod, hetresselector, waterselector))
+        println(io, "Number of atoms             -  ", countatoms(mod, stdatomselector))
+        println(io, "Number of hydrogens         -  ", countatoms(mod, stdatomselector, hydrogenselector))
+        println(io, "Number of disordered atoms  -  ", countatoms(mod, stdatomselector, disorderselector))
```

Comment:
Printing a newline at the last line will print an extra newline in REPL. I think `print(io, "Number of...."))` would be better.